src/components/core/Button/Button.tsx

```diff
@@ -1,107 +1,122 @@
-import { memo, useMemo } from 'react';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 import Loader from '../Loader/Loader';
-import { TitleXS } from '../Text/Text';
 import colors from '../colors';
 import transitions from '../transitions';
-
-type ButtonStyle = 'primary' | 'secondary';
-
-type Props = {
-  type?: ButtonStyle;
-  className?: string;
-  text?: string;
-  mini?: boolean;
-  onClick?: () => void;
-  disabled?: boolean;
-  loading?: boolean;
-  dataTestId?: string;
-};
-
-function Button({
-  type = 'primary',
-  className,
-  text,
-  mini = false,
-  onClick,
-  disabled,
-  loading,
-  dataTestId,
-}: Props) {
-  const ButtonComponent = useMemo(
-    () => (type === 'primary' ? StyledPrimaryButton : StyledSecondaryButton),
-    [type]
-  );
-  return (
-    <ButtonComponent
-      // Renaming this prop `buttonStyle` since the `type` prop
-      // already exists for styled components
-      buttonStyle={type}
-      className={className}
-      onClick={onClick}
-      disabled={disabled}
-      data-testid={dataTestId}
-      mini={mini}
-    >
-      {loading ? (
-        <Loader inverted size={mini ? 'mini' : 'small'} />
-      ) : (
-        <TitleXS color={type === 'primary' ? colors.white : colors.shadow}>{text}</TitleXS>
-      )}
-    </ButtonComponent>
-  );
-}
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react';
+import { BODY_FONT_FAMILY } from '../Text/Text';
+import Link from 'next/link';
+
+// TODO:
+// - should a `loading` button be disabled/non-interactive?
+// - should `Loader` be conditionally inverted based on `variant`?
+// - should `Loader` just always be `mini`?
+// - why is opacity on disabled+loading so much stronger than just disabled?
+// - does ButtonLink need to handle absolute URLs separately? or does Link do that?
+//   - should ButtonLink also add rel="noopener,noreferer" to target="_blank" links?
 
 type StyledButtonProps = {
+  variant?: 'primary' | 'secondary';
   disabled?: boolean;
-  buttonStyle: ButtonStyle;
-  mini?: boolean;
 };
 
 const StyledButton = styled.button<StyledButtonProps>`
   display: flex;
   justify-content: center;
   align-items: center;
-
-  border-style: none;
-
-  height: 32px;
-
+  border: 0;
   cursor: pointer;
 
-  text-transform: uppercase;
   padding: 0 24px;
+  height: 32px;
```

Comment:
I don't love fixing the height of a button like this. I tried to move this to be top/bottom padding, but the loading indicator throws this off (it's >1em).

Ideally the loading indicator is 1em so that it matches the text size or, if it needs to be larger, it includes negative margins to avoid the button growing/shrinking in height as `loading` is toggled.